src/measure_theory/prod.lean

```diff
@@ -0,0 +1,689 @@
+lemma measure_prod_null {s : set (α × β)}
+  (hs : is_measurable s) : μ.prod ν s = 0 ↔ (λ x, ν (prod.mk x ⁻¹' s)) =ᵐ[μ] 0 :=
+by simp_rw [prod_apply hs, lintegral_eq_zero_iff (measurable_measure_prod_mk_left hs)]
+
+/-- Note: the converse is not true without assuming that `s` is measurable. For a counterexample,
+  see Walter Rudin *Real and Complex Analysis*, example (c) in section 8.9. -/
+lemma measure_ae_null_of_prod_null {s : set (α × β)}
+  (h : μ.prod ν s = 0) : (λ x, ν (prod.mk x ⁻¹' s)) =ᵐ[μ] 0 :=
+begin
+  obtain ⟨t, hst, mt, ht⟩ := exists_is_measurable_superset_of_measure_eq_zero h,
+  simp_rw [measure_prod_null mt] at ht,
+  rw [eventually_le_antisymm_iff],
+  exact ⟨eventually_le.trans_eq
+    (eventually_of_forall $ λ x, (measure_mono (preimage_mono hst) : _)) ht,
+    eventually_of_forall $ λ x, zero_le _⟩
+end
+
+/-- Note: the converse is not true. For a counterexample, see
+  Walter Rudin *Real and Complex Analysis*, example (c) in section 8.9. -/
+lemma ae_ae_of_ae_prod {p : α × β → Prop} (h : ∀ᵐ z ∂μ.prod ν, p z) :
+  ∀ᵐ x ∂ μ, ∀ᵐ y ∂ ν, p (x, y) :=
+measure_ae_null_of_prod_null h
+
+variables [sigma_finite μ]
+
+instance prod.sigma_finite : sigma_finite (μ.prod ν) :=
+⟨⟨λ n, (spanning_sets μ n).prod (spanning_sets ν n),
+  λ n, (is_measurable_spanning_sets μ n).prod (is_measurable_spanning_sets ν n),
+  λ n, by { simp_rw [prod_prod (is_measurable_spanning_sets μ n) (is_measurable_spanning_sets ν n)],
+    exact mul_lt_top (measure_spanning_sets_lt_top μ n) (measure_spanning_sets_lt_top ν n) },
+  by { simp_rw [Union_prod_of_monotone (monotone_spanning_sets μ) (monotone_spanning_sets ν),
+    Union_spanning_sets, univ_prod_univ] }⟩⟩
+
+/- Note: This proof would be shorter if `sigma_finite` was not `Prop`-valued, since we use that
+  the sets given in the instance of `sigma_finite` is a π-system. -/
+/-- Measures on a product space are equal if they are equal on rectangles. -/
+lemma prod_unique {μν₁ μν₂ : measure (α × β)}
+  (h₁ : ∀ s t, is_measurable s → is_measurable t → μν₁ (s.prod t) = μ s * ν t)
+  (h₂ : ∀ s t, is_measurable s → is_measurable t → μν₂ (s.prod t) = μ s * ν t) : μν₁ = μν₂ :=
+begin
+  refine ext_of_generate_from_of_Union _
+    (λ i, (spanning_sets μ i).prod (spanning_sets ν i))
+    generate_from_prod.symm is_pi_system_prod _ _ _ _,
+  { rw [Union_prod_of_monotone (monotone_spanning_sets μ) (monotone_spanning_sets ν)],
+    simp_rw [Union_spanning_sets, univ_prod_univ] },
+  { intro i, apply mem_image2_of_mem; apply is_measurable_spanning_sets },
+  { intro i, rw [h₁], apply mul_lt_top; apply measure_spanning_sets_lt_top,
+    all_goals { apply is_measurable_spanning_sets } },
+  { rintro _ ⟨s, t, hs, ht, rfl⟩, simp * at * }
+end
+
+lemma prod_eq {μν : measure (α × β)}
+  (h : ∀ s t, is_measurable s → is_measurable t → μν (s.prod t) = μ s * ν t) : μ.prod ν = μν :=
+prod_unique (λ s t hs ht, prod_prod hs ht) h
+
+lemma prod_swap : map prod.swap (μ.prod ν) = ν.prod μ :=
```

Comment:
Two more theorems I'd expect to have: `map (prod.map f g) (μ.prod ν) = (map f μ).prod (map g ν)`, `map (equiv.prod_assoc α β γ) ((μa.prod μb).prod μc) = μa.prod (μb.prod μc)`.